phac-nml/irida-next
# Update: Group samples count calculation should include shared groups and projects

## What does this PR do and why?
_Describe in detail what your merge request does and why._
This PR changes the sample count of a group to incorporate sample counts from it's shared groups and projects. This update will allow the samples count to properly reflect the amount of samples shown after clicking samples on the sidebar for a group.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._
General check:
1. Click on Groups in the sidebar.
2. Take note of the samples count for any group that you know has a group or project shared with it. Click on that group.
3. Click on Samples in the sidebar.
4. At the bottom of the page, there is a section saying "Display 1-20 of X items" where X is the total number of samples this group has access to. Confirm that this number equals the one noted in step 2.

Sharing a project:
1. Click on Groups in the sidebar.
2. Take note of the samples count for any group that you have access to.
3. Click on Projects in the sidebar.
4. Select a project that you own, that has not already been shared to the group in step 2.
5. Click on Members in the sidebar.
6. Select the "Invite group" button.
7. Select the group that was noted in step 2, and give it an access level of Analyst. Click "Share".
8. Repeat the general check above with the group selected in step 2. The samples count should have increase by the direct samples count** of the project we shared. 

Sharing a group:
1. Click on Groups in the sidebar.
2. Take note of the samples count for any group that you have access to.
3. Select a different group that you have access to and want to share with the group in step 2. This different group should not already be shared with the group in step 2.
5. Click on Members in the sidebar.
6. Select the "Invite group" button.
7. Select the group that was noted in step 2, and give it an access level of Analyst. Click "Share".
8. Repeat the general check above with the group selected in step 2. The samples count should have increase by the direct samples count** of the group we shared. 

**Direct samples count for a shared project = samples count of that project
**Direct samples count for a shared group = the sum of all samples found within projects in itself and it's ancestor groups. (This does not include the samples count of projects or groups that have been shared with it)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
